clang/lib/3C/CastPlacement.cpp

```diff
@@ -66,6 +66,10 @@ bool CastPlacementVisitor::VisitCallExpr(CallExpr *CE) {
 
       CVarSet ArgConstraints = CR.getExprConstraintVarsSet(ArgExpr);
       for (auto *ArgC : ArgConstraints) {
+        // If the function takes a void *, we already know about the wildness,
+        // so allow the implicit cast.
+        if (TypeVar == nullptr && FV->getExternalParam(PIdx)->isVoidPtr())
```

Comment:
It looks like this was added to avoid inserting an unnecessary casts, but I can't think of an example that would trigger this code path. Do you have an example/test case that could be added?

Reply:
The original example ( first block in the test file here ) gets a cast to void* if this code is removed. Because it now has constraints generated without the implicit cast, an explicit one is added. So this code prevents that.